components/suggestion/command_provider/app_command_parser/app_command_parser.ts

```diff
@@ -1463,7 +1470,14 @@ export class AppCommandParser {
             return {form: submittableForm};
         }
 
-        const fetched = await this.fetchForm(binding);
+        if (!binding.form?.source) {
+            return {error: this.intl.formatMessage({
+                id: 'apps.error.parser.missing_source',
+                defaultMessage: 'Form has neither submit nor source.',
+            })};
+        }
+        const context = this.getAppContext(binding);
+        const fetched = await this.fetchSubmittableForm(binding.form?.source, context);
```

Comment:
Nit: `binding.form` doesn't need the question mark on this line since we already checked above, so typescript infers that `source` is defined.

```suggestion
        const fetched = await this.fetchSubmittableForm(binding.form.source, context);
```